src/pages/api/auth/signup.ts

```diff
@@ -4,29 +4,70 @@ import nextConnect from 'next-connect';
 import bcrypt from 'bcrypt';
 import moment from 'moment';
 import pool from 'utils/db';
+import { createSupplier, createBuyer } from './user';
+
+const checkEmaiExist = async (Email: string) => {
+  const result = await pool.query(`
+    SELECT EXISTS (SELECT * FROM "User" WHERE "Email"='${Email}');`);
+  return result.rows[0].exists;
+};
 
 const handler = nextConnect().post(
-  (req: NextApiRequest, res: NextApiResponse) => {
-    const { Password, Name, Email } = req.body;
-    // const { Password, Name, Email, Type } = req.body;
-    // if(Type === 'supplier'){
-    //   // Add Supplier
-    // }
+  async (req: NextApiRequest, res: NextApiResponse) => {
+    const { Type, Password, Name, Email, ABN, Logo } = req.body;
+
+    const exists = await checkEmaiExist(Email);
+    if (exists) {
+      res
+        .status(400)
+        .json({ success: false, message: 'Email is already exist!' });
+      return;
+    }
 
     const hashedPassword = bcrypt.hashSync(Password as string, 10);
-    pool
-      .query(
-        `INSERT INTO "User" ("Name", "Password", "Email", "CreatedAt") VALUES ('${Name}', '${hashedPassword}', '${Email}', '${moment().format(
-          'YYYY-MM-DD',
-        )}');`,
-      )
-      .then((mes: any) => {
-        if (mes) res.status(200).json({ success: true });
-      })
-      .catch((err: any) => {
-        console.error(err);
-        if (err) res.status(500).json(err);
-      });
+    if (Type === 'supplier') {
+      const { State_ID, City_ID } = req.body;
+      const SupplierID = await createSupplier(
+        Name,
+        ABN,
+        Logo,
+        State_ID,
+        City_ID,
+        res,
+      );
+
+      pool
+        .query(
+          `INSERT INTO "User" ("Name", "Password", "Email", "Supplier_ID", "CreatedAt") VALUES ('${Name}', '${hashedPassword}', '${Email}', '${SupplierID}', '${moment().format(
+            'YYYY-MM-DD',
+          )}');`,
+        )
```

Comment:
since you started using await the code should be
```ts
const mes = await pool
        .query(
          `INSERT INTO "User" ("Name", "Password", "Email", "Supplier_ID", "CreatedAt") VALUES ('${Name}', '${hashedPassword}', '${Email}', '${SupplierID}', '${moment().format(
            'YYYY-MM-DD',
          )}');`,
        )
if(mes){
   res.status(200).json({ success: true })
}
```